beacon-chain/rpc/prysm/v1alpha1/validator/proposer_attestations.go

```diff
@@ -25,13 +26,18 @@ func (a proposerAtts) filter(ctx context.Context, st state.BeaconState) (propose
 	validAtts := make([]*ethpb.Attestation, 0, len(a))
 	invalidAtts := make([]*ethpb.Attestation, 0, len(a))
 	var attestationProcessor func(context.Context, state.BeaconState, *ethpb.Attestation) (state.BeaconState, error)
+
 	switch st.Version() {
 	case version.Phase0:
 		attestationProcessor = blocks.ProcessAttestationNoVerifySignature
 	case version.Altair:
 		// Use a wrapper here, as go needs strong typing for the function signature.
 		attestationProcessor = func(ctx context.Context, st state.BeaconState, attestation *ethpb.Attestation) (state.BeaconState, error) {
-			return altair.ProcessAttestationNoVerifySignature(ctx, st, attestation)
+			totalBalance, err := helpers.TotalActiveBalance(st)
```

Comment:
remove this outside the declared function. That way it is only done once, otherwise the end result would be the same as before.

Reply:
Correct. Fully aware of this. This is for proposer proposing. Moving it outside will require me changing:

`filter(ctx context.Context, st state.BeaconState) (proposerAtts, proposerAtts)`
to
`filter(ctx context.Context, st state.BeaconState) (proposerAtts, proposerAtts, error)`

I wasn't too keen on doing that so I think this could be solved by the cache 😅